examples/keras_io/tensorflow/nlp/ner_transformers.py

```diff
@@ -0,0 +1,341 @@
+Last modified: Jun 24, 2021
+Description: NER using the Transformers and data from CoNLL 2003 shared task.
+Accelerator: GPU
+"""
+"""
+## Introduction
+
+Named Entity Recognition (NER) is the process of identifying named entities in text.
+Example of named entities are: "Person", "Location", "Organization", "Dates" etc. NER is
+essentially a token classification task where every token is classified into one or more
+predetermined categories.
+
+In this exercise, we will train a simple Transformer based model to perform NER. We will
+be using the data from CoNLL 2003 shared task. For more information about the dataset,
+please visit [the dataset website](https://www.clips.uantwerpen.be/conll2003/ner/).
+However, since obtaining this data requires an additional step of getting a free license, we will be using
+HuggingFace's datasets library which contains a processed version of this dataset.
+"""
+
+"""
+## Install the open source datasets library from HuggingFace
+
+We also download the script used to evaluate NER models.
+"""
+
+"""shell
+pip3 install datasets
+wget https://raw.githubusercontent.com/sighsmile/conlleval/master/conlleval.py
+"""
+
+import os
+import numpy as np
+import tensorflow as tf
+import keras_core as keras
+from keras_core import layers
+from datasets import load_dataset
+from collections import Counter
+from conlleval import evaluate
+
+
+class TransformerBlock(layers.Layer):
+    def __init__(self, embed_dim, num_heads, ff_dim, rate=0.1):
+        super().__init__()
+        self.att = keras.layers.MultiHeadAttention(
+            num_heads=num_heads, key_dim=embed_dim
+        )
+        self.ffn = keras.Sequential(
+            [
+                keras.layers.Dense(ff_dim, activation="relu"),
+                keras.layers.Dense(embed_dim),
+            ]
+        )
+        self.layernorm1 = keras.layers.LayerNormalization(epsilon=1e-6)
+        self.layernorm2 = keras.layers.LayerNormalization(epsilon=1e-6)
+        self.dropout1 = keras.layers.Dropout(rate)
+        self.dropout2 = keras.layers.Dropout(rate)
+
+    def call(self, inputs, training=False):
+        attn_output = self.att(inputs, inputs)
+        attn_output = self.dropout1(attn_output, training=training)
+        out1 = self.layernorm1(inputs + attn_output)
+        ffn_output = self.ffn(out1)
+        ffn_output = self.dropout2(ffn_output, training=training)
+        return self.layernorm2(out1 + ffn_output)
+
+
+"""
+Next, let's define a `TokenAndPositionEmbedding` layer:
+"""
+
+
+class TokenAndPositionEmbedding(layers.Layer):
+    def __init__(self, maxlen, vocab_size, embed_dim):
+        super().__init__()
+        self.token_emb = keras.layers.Embedding(
+            input_dim=vocab_size, output_dim=embed_dim
+        )
+        self.pos_emb = keras.layers.Embedding(input_dim=maxlen, output_dim=embed_dim)
+
+    def call(self, inputs):
+        maxlen = tf.shape(inputs)[-1]
+        positions = tf.range(start=0, limit=maxlen, delta=1)
```

Comment:
Suggestion: could be replaced with `keras_cors.ops.arange`

Reply:
changed the parameter names according to `keras_core.ops.arange`